zk/conn.go

```diff
@@ -317,6 +326,65 @@ func (c *Conn) connect() error {
 	}
 }
 
+func (c *Conn) resendZkAuth(reauthReadyChan chan struct{}) {
+
+	c.credsMu.Lock()
+	if len(c.creds) > 0 {
```

Comment:
actually no need for the if... this code will just work. just move the log message into the loop